ui/apps/profile/src/components/pages/profile-info/index.tsx

```diff
@@ -47,11 +48,20 @@ const ProfileInfoPage: React.FC<ProfileInfoPageProps> = props => {
   const { akashaProfile: profileData } =
     data?.node && hasOwn(data.node, 'akashaProfile') ? data.node : { akashaProfile: null };
 
-  const [showNSFW, setShowNSFW] = useState(false);
-
   const isLoggedIn = !!loginData?.id;
   const authenticatedDID = loginData?.id;
 
+  const { showNsfw: showNsfwSetting } = useNsfwToggling();
+  const [showNSFW, setShowNSFW] = useState(false);
+
+  /* show the overlay or display the profile directly depending on the user's nsfw setting
+  and whether they are logged in or not */
+  React.useEffect(() => {
+    if (isLoggedIn && showNsfwSetting) {
```

Comment:
if a logged out user has nsfw content hidden by default, and only if a user logs in can he change the nsfw settings, then maybe you can add this condition of a logged in user in the useNsfwToggling hook to simplify its usage